src/containers/Tenant/Query/QuerySettingsDialog/QuerySettingsDialog.tsx

```diff
@@ -7,65 +7,43 @@ import {
     selectQueryAction,
     setQueryAction,
 } from '../../../../store/reducers/queryActions/queryActions';
-import type {
-    IsolationLevel,
-    QueryMode,
-    StatisticsMode,
-    TracingLevel,
-} from '../../../../types/store/query';
+import type {QuerySettings} from '../../../../types/store/query';
 import {cn} from '../../../../utils/cn';
-import {useTypedDispatch, useTypedSelector} from '../../../../utils/hooks';
 import {
-    ISOLATION_LEVELS,
-    QUERY_MODES,
-    STATISTICS_MODES,
-    TRACING_LEVELS,
-} from '../../../../utils/query';
+    useQueryExecutionSettings,
+    useTypedDispatch,
+    useTypedSelector,
+} from '../../../../utils/hooks';
 
 import {QuerySettingsSelect} from './QuerySettingsSelect';
-import {
-    ISOLATION_LEVEL_SELECT_OPTIONS,
-    QUERY_MODE_SELECT_OPTIONS,
-    STATISTICS_MODE_SELECT_OPTIONS,
-    TRACING_LEVEL_SELECT_OPTIONS,
-} from './constants';
+import {QUERY_SETTINGS_FIELD_SETTINGS} from './constants';
 import i18n from './i18n';
 
 import './QuerySettingsDialog.scss';
 
 const b = cn('ydb-query-settings-dialog');
 
-type FormValues = {
-    queryMode: QueryMode;
-    timeout: string;
-    isolationLevel: IsolationLevel;
-    statisticsMode: StatisticsMode;
-    tracingLevel: TracingLevel;
-};
-
 export function QuerySettingsDialog() {
     const dispatch = useTypedDispatch();
     const queryAction = useTypedSelector(selectQueryAction);
-    const {control, handleSubmit, reset} = useForm<FormValues>({
-        defaultValues: {
-            queryMode: QUERY_MODES.script,
-            timeout: '60',
-            isolationLevel: ISOLATION_LEVELS.serializable,
-            statisticsMode: STATISTICS_MODES.none,
-            tracingLevel: TRACING_LEVELS.detailed,
-        },
+    const [querySettings, setQuerySettings] = useQueryExecutionSettings();
+    const {control, handleSubmit, reset} = useForm<QuerySettings>({
+        defaultValues: querySettings,
     });
 
-    const onCloseDialog = () => {
+    const onCloseDialog = React.useCallback(() => {
```

Comment:
onCloseDialog is passed to Dialog component and we don't want it to rerender every time onCloseDialog function is reinitialized. Also it is passed to Dialog.Footer and we don't actually know how it may be used there - any useEffect that depends on onCloseDialog will fire every time our QuerySettingsDialog is rerendered

so it is always good practice to use useCallback to functions that are passed to child components as props